### PR TITLE
Fix EventValue initialization from ctor

### PR DESCRIPTION
**Pull request by Ky7m**

### CODE REVIEW

**Comment by tomlm** [comment]
>         public EmitEvent(string eventName = null, object eventValue = null, bool bubble = true, [CallerFilePath] string callerPath = "", [CallerLineNumber] int callerLine = 0)

this should be  a string, eventValue is intented to be a property path to memory that is evaluated when the EmitEvent is executed

---
Refers to: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/EmitEvent.cs:19 in efe2522. [](commit_id = efe2522132b1786fc20e38a82847e594d86f1858, deletion_comment = False)

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/EmitEvent.cs [hunk 1]
```diff
@@ -21,7 +21,7 @@ public EmitEvent(string eventName = null, object eventValue = null, bool bubble
         {
             this.RegisterSourceLocation(callerPath, callerLine);
             this.EventName = eventName;
-            this.EventValue = EventValue;
+            this.EventValue = eventValue;
```
Comment (tomlm) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/EmitEvent.cs]: >eventValue [](start = 30, length = 10)

This is right, nice catch!

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/EmitEvent.cs [hunk 2]
```diff
@@ -33,7 +33,7 @@ public EmitEvent(string eventName = null, object eventValue = null, bool bubble
         /// <summary>
         /// Gets or sets the object to send with the event
         /// </summary>
-        public string EventValue { get; set; }
+        public object EventValue { get; set; }
```
Comment (tomlm) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/EmitEvent.cs]: >object [](start = 15, length = 6)

this should not change, it should be a string

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/EmitEvent.cs [hunk 3]
```diff
@@ -33,7 +33,7 @@ public EmitEvent(string eventName = null, object eventValue = null, bool bubble
         /// <summary>
         /// Gets or sets the object to send with the event
```
Comment (tomlm) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/EmitEvent.cs]: >object [](start = 29, length = 6)

"Gets or sets the memory property path to use to get the value to send as part of the event"

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/EmitEvent.cs [hunk 4]
```diff
@@ -47,7 +47,7 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
                 throw new ArgumentException($"{nameof(options)} cannot be a cancellation token");
             }
 
-            var eventValue = (this.EventValue != null) ? dc.State.GetValue<object>(this.EventValue) : null;
+            var eventValue = (this.EventValue != null) ? dc.State.GetValue<object>(EventValuePropertyValue) : null;
```
Comment (tomlm) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/EmitEvent.cs]: >EventValuePropertyValue [](start = 83, length = 23)

no, this should be EventValue as a string

**Comment by tomlm** [comment]
>         private const string EventValuePropertyValue = "eventValue";

I think we should remove this.

---
Refers to: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/EmitEvent.cs:16 in efe2522. [](commit_id = efe2522132b1786fc20e38a82847e594d86f1858, deletion_comment = False)

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/EmitEvent.cs (hunk 1 above)
Comment (Ky7m) [reply libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/EmitEvent.cs]: thanks

**Comment by Ky7m** [comment]
thank you @tomlm for comments. Fixed now.

**Pull request merged**